common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1864,35 +1871,27 @@ private void finalizePartition() throws IOException {
       metaFile.getChannel().truncate(metaFile.getPos());
     }
 
-    void closeAllFilesAndDeleteIfNeeded(boolean delete) {
+    private void deleteAllFiles() {
       try {
-        if (dataChannel.isOpen()) {
-          dataChannel.close();
-        }
-        if (delete) {
-          dataFile.delete();
-        }
-      } catch (IOException ioe) {
-        logger.warn("Error closing data channel for {} reduceId {}",
+        if (!dataFile.delete()) {
+          logger.warn("Error deleting data file for {} reduceId {}",
             appAttemptShuffleMergeId, reduceId);
+        }
+      } catch (Exception exception) {
+        logger.warn("Error deleting data file for {} reduceId {}",
+          appAttemptShuffleMergeId, reduceId);
```

Comment:
We dont need to handle security manager related exceptions, they are not relevant for spark - we can drop this catch block.